# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../gdb/legacy_sus.patch
# Copyright (C) 2010 - 2012 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

usleep() was deprecated by SUSv3 and newer LIBCs are starting to enforce that
deprecation.

diff --git a/gdb/linux-nat.c b/gdb/linux-nat.c
index 44a2a21..4e70f80 100644
--- a/gdb/linux-nat.c
+++ b/gdb/linux-nat.c
@@ -66,6 +66,7 @@
 #include "exceptions.h"
 #include "linux-ptrace.h"
 #include "buffer.h"
+#include <time.h>		/* for nanosleep */
 
 #ifndef SPUFS_MAGIC
 #define SPUFS_MAGIC 0x23c9b64e
@@ -841,7 +842,8 @@
 				    "LCFF: no VFORK_DONE "
 				    "support, sleeping a bit\n");
 
-	      usleep (10000);
+	      const struct timespec t = {0, 10000000L};
+	      nanosleep(&t, NULL);
 
 	      /* Pretend we've seen a PTRACE_EVENT_VFORK_DONE event,
 		 and leave it pending.  The next linux_nat_resume call
